circle.yaml

```diff
@@ -0,0 +1,22 @@
+machine:
+    services:
+        - docker
+    environment:
+        CGO_ENABLED: "0"
+        GOOS: "linux"
+        GOARCH: "amd64"
+        GOPATH: "$HOME/go"
+        GS_WD: "$HOME/go/src/github.com/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME"
+
+dependencies:
+    override:
+        - sudo add-apt-repository ppa:masterminds/glide -y
+        - sudo apt-get update
+        - sudo apt-get install glide -y
+        - mkdir -p $HOME/go/src/github.com/$CIRCLE_PROJECT_USERNAME
+        - cp -rf $HOME/$CIRCLE_PROJECT_REPONAME $HOME/go/src/github.com/$CIRCLE_PROJECT_USERNAME
+
+test:
+    override:
+        - cd $GS_WD && go test -v $(glide novendor)
+        - cd $GS_WD && go build -a -v -tags netgo -ldflags "-w -X main.gitCommit=$(git rev-parse --short HEAD)"
```

Comment:
Mr. @JosephSalisbury has an opinion here. I think we want to use architect here. See https://github.com/giantswarm/api/blob/master/circle.yml.